### PR TITLE
Migrations hardening database tables

### PR DESCRIPTION
**Pull request by JamesMGreene**
Follow-up to:
- https://github.com/ProgramEquity/amplify-back-end/pull/138

In order to get our production and local databases into parity, we had to drop some of our hardening constraints from the local databases temporarily (via PR #138 ☝🏻). This PR reintroduces all of those constraints and will be applied across both the local **and** production databases once merged. ✨

### CODE REVIEW

File: server/db/migrations/20211230165827_harden-letter_versions-table.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const tableName = 'letter_versions'
+
+module.exports = {
+  async up(knex) {
+    await knex.schema.alterTable(tableName, function (table) {
+      // Rename columns
+      table.renameColumn('campaignid', 'campaign_id')
```
Comment (nawazkhan) [inline server/db/migrations/20211230165827_harden-letter_versions-table.js:7]: Sweet! After this update, we may need to update this [line](https://github.com/ProgramEquity/amplify-back-end/blob/main/server/routes/api/letter_versions.js#L9) cc: @BeeXiong

File: server/db/migrations/20211230165827_harden-letter_versions-table.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      table.string('template_id').notNullable().alter()
+
+      // Add columns
+      table.string('municipality')
+
+      // Add indexes
+      table.index(['campaign_id'])
+
+      // Add unique indexes
+      table.unique(['template_id'])
```
Comment (nawazkhan) [inline server/db/migrations/20211230165827_harden-letter_versions-table.js:20]: Is unique required here? I think we can add it later if needed.

**Review by nawazkhan — APPROVED** [review]
LG!

File: server/db/migrations/20211230165827_harden-letter_versions-table.js (hunk 1 above)
Comment (manishapriya94) [reply server/db/migrations/20211230165827_harden-letter_versions-table.js:7]: @BeeXiong any thoughts here?

Comment (JamesMGreene) [reply server/db/migrations/20211230165827_harden-letter_versions-table.js:7]: Updated in commit https://github.com/ProgramEquity/amplify-back-end/pull/147/commits/27aa8632b01ed8058caa16a5cc76b37ab95692a6

File: server/db/migrations/20211230165827_harden-letter_versions-table.js (hunk 2 above)
Comment (JamesMGreene) [reply server/db/migrations/20211230165827_harden-letter_versions-table.js:20]: My understanding is that they should always be unique, so this gives us uniqueness enforcement + indexing. 🤷🏻‍♂️

**Pull request merged**